src/pages/post-page.html

```diff
@@ -0,0 +1,163 @@
+      }
+
+      [slot="markdown-html"] h1,
+      [slot="markdown-html"] h2,
+      [slot="markdown-html"] h3 {
+        margin: 48px 0 16px;
+      }
+
+      [slot="markdown-html"] p {
+        margin-top: 0;
+        margin-bottom: 24px;
+      }
+
+      [slot="markdown-html"] plastic-image {
+        margin: 32px 0 8px -16px;
+        --iron-image-width: calc(100% + 32px);
+        width: calc(100% + 32px);
+        min-height: 200px;
+        background-color: var(--secondary-background-color);
+      }
+
+      .date {
+        font-size: 12px;
+        text-transform: uppercase;
+        color: var(--secondary-text-color);
+      }
+
+      .suggested-posts {
+        margin: 24px 0 -20px;
+        padding-top: 24px;
+        background-color: var(--primary-background-color);
+      }
+
+      @media (min-width: 640px) {
+        [slot="markdown-html"] plastic-image {
+          min-height: 400px;
+        }
+
+        .suggested-posts {
+          margin-top: 48px;
+        }
+      }
+
+    </style>
+
+    <app-route
+      route="[[route]]"
+      pattern="/:id"
+      data="{{postData}}"
+    ></app-route>
+
+    <iron-ajax
+      url="[[source]]"
+      handle-as="text"
+      last-response="{{postBody}}"
+      auto
+    ></iron-ajax>
+
+    <div class="content content-narrow">
+      <marked-element class="post" markdown="[[postBody]]">
+        <div slot="markdown-html"></div>
+      </marked-element>
+      <div class="date">{$ blog.published $}: [[published]]</div>
+    </div>
+
+    <div class="suggested-posts">
+      <div class="content content-narrow">
+        <h3 class="content-title">{$ blog.suggested $}</h3>
+        <posts-list posts="[[suggestedPosts]]"></posts-list>
+      </div>
+    </div>
+
+  </template>
+
+  <script>
+
+    class PostPage extends UtilsFunctions(ReduxMixin(Polymer.Element)) {
+      static get is() {
+        return 'post-page';
+      }
+
+      static get properties() {
+        return {
+          route: Object,
+
+          postsList: {
+            type: String,
+            statePath: 'blog.list'
+          }
+        };
+      }
+
+      static get observers() {
+        return [
+          '_postDataObserver(postData.id, postsList)'
+        ];
+      }
+
+      connectedCallback() {
+        super.connectedCallback();
+        if (!Object.keys(this.postsList).length) return blogActions.fetchList();
+      }
+
+      _postDataObserver(postId, postsList) {
+        if (!this.postsList || !Object.keys(this.postsList).length) return;
+        this._setHero(this.postsList[this.postData.id]);
+        this.set('source', this.postsList[this.postData.id].source);
+        this.set('published', this._getDate(this.postsList[this.postData.id].published));
+        const sortedPosts = this.transformToArray(this.postsList, 'id').sort((a, b) => b.published.localeCompare(a.published));
+        this.set('suggestedPosts', sortedPosts.filter(post => post.id !== this.postData.id).slice(0, 3));
```

Comment:
Yes, but sometimes we also have to think about logical units, and the function does only 1 job, IMHO